eis_toolkit/raster_processing/unique_combinations.py

```diff
@@ -0,0 +1,75 @@
+from typing import Dict, List, Tuple
+
+import numpy as np
+import rasterio
+
+from eis_toolkit.exceptions import InvalidParameterValueException
+
+
+def _unique_combinations(
+    bands: List[np.ndarray],
+) -> np.ndarray:
+
+    unique_combinations: Dict[int, int] = {}
+    unique_indices = np.zeros(bands[0].shape, dtype=int)
+    combination = 1
+
+    for band in bands:
+        for row in range(band.shape[0]):
+            for column in range(band.shape[1]):
+                raster_value = band[row, column]
+                if raster_value not in unique_combinations:
+                    unique_combinations[raster_value] = combination
+                    combination += 1
+                unique_indices[row, column] = unique_combinations[raster_value]
```

Comment:
I'm having a little trouble trying to verify this works as intended. Does this consider combinations across all bands right now? It seems it might only check if individual value was encountered previously. If this is the case, you could leave looping of the bands out and inside the row-col loop do something like this:

`combination = tuple(band[row, col] for band in bands)`
`unique_indices[row, col] = unique_combinations[combination]`

Also while looking at this PR I happened to find this neat tool you could maybe use here: `defaultdict`.

Reply:
@nmaarnio I have now used numpy to get the unique combinations. It is faster with a 10k x 10k grid compared to the old one (95 sec to 85 sec). I apologize for the bad naming of the commits. I made commits called 'Use numpy to find unique combinations without nested loops'. Somehow, with my lack of understanding of pre-commit, the commits got lost in the way.